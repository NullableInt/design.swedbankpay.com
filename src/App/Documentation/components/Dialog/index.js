import React, { Component } from "react";

import { ComponentPreview, DocContainer, Icon, Property } from "#";
import { dialog } from "$/px-script/main";

const Example = () => (
    <>
        <h2 id="example">Example</h2>
        <p>Example dialog...</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="dialog" id="demo-dialog">
                <section>
                    <header>
                        <h5>Delete item 456?</h5>{"\n"}
                        <a href="#" className="dialog-close">{"\n\t\t\t\t"}
                            <Icon icon="close" />{"\n\t\t\t"}
                        </a>{"\n\t\t"}
                    </header>
                    <div className="dialog-body">
                        <p>Are you sure you want to permanently delete the item <i>German Swashbuckle (456)?</i></p>
                    </div>
                    <footer>{"\n"}
                        <button className="btn btn-secondary" type="button" data-dialog-close="demo-dialog">Cancel</button>{"\n"}
                        <button className="btn btn-danger" type="button">Delete</button>{"\n"}
                    </footer>
                </section>
            </div>{"\n"}
            <button className="btn btn-primary" type="button" data-dialog-open="demo-dialog">Open dialog</button>
        </ComponentPreview>
    </>
);

const UsageWithJavascript = () => (
    <>
        <h2 id="usage-with-javascript">Usage with javascript</h2>
        <p>To programmatically open a dialog, use <Property value={"px.dialog.open(\"DIALOG_ID\")"} />.</p>
        <p>To programmatically close a dialog, use <Property value={"px.dialog.close(\"DIALOG_ID\")"} />.</p>
    </>
);

class Dialog extends Component {
    componentDidMount () {
        dialog.init();
    }

    render () {
        return (
            <DocContainer docToc>
                <p className="lead">Dialogs...</p>
                <Example />
                <UsageWithJavascript />
            </DocContainer>
        );
    }
}

export default Dialog;

/* For testing */
export { Example, UsageWithJavascript };
