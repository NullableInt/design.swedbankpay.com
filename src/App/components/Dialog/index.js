import React from "react";
import PropTypes from "prop-types";

import { Icon } from "#";

const Dialog = ({ diaId, diaHeader, children }) => (
    <>
        <div className="dialog" id={diaId} role="dialog" aria-modal="true">
            <section>
                <header className="dialog-header">
                    <h4>{diaHeader}</h4>
                    <button type="button" className="dialog-close">
                        <Icon type="close" />
                    </button>
                </header>
                <div className="dialog-body">
                    {children}
                </div>
                <footer className="dialog-footer">
                    <button className="btn btn-guiding" type="button" data-dialog-close>Cancel</button>{"\n"}
                    <button className="btn btn-destructive" type="button">Delete</button>{"\n"}
                </footer>
            </section>
        </div>
    </>
);

Dialog.propTypes = {
    diaId: PropTypes.string,
    diaHeader: PropTypes.string
};

export default Dialog;
