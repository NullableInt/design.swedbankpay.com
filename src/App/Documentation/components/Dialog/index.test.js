import React from "react";
import { shallow } from "enzyme";

import Dialog, { HowItWorks, Example, UsageWithJavascript } from "./index";

describe("Documentation: Dialog", () => {
    it("is defined", () => {
        expect(Dialog).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Dialog />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("HowItWorks", () => {
        it("is defined", () => {
            expect(HowItWorks).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<HowItWorks />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("Example", () => {
        it("is defined", () => {
            expect(Example).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Example />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("UsageWithJavascript", () => {
        it("is defined", () => {
            expect(UsageWithJavascript).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<UsageWithJavascript />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
