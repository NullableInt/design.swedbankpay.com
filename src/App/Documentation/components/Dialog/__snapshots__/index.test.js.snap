// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation: Dialog Example renders 1`] = `
<Fragment>
  <h2
    id="overview"
  >
    Example
  </h2>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <button
      className="btn btn-primary"
      data-dialog-open="demo-dialog"
      type="button"
    >
      Open dialog
    </button>
    <div
      className="dialog"
      id="demo-dialog"
    >
      <section>
        <header>
          <h5>
            Delete item 456?
          </h5>
          

          <a
            className="dialog-close"
            href="#"
          >
            
				
            <Icon
              icon="close"
            />
            
			
          </a>
          
		
        </header>
        <div
          className="dialog-body"
        >
          <p>
            Are you sure you want to permanently delete the item 
            <i>
              German Swashbuckle (456)?
            </i>
          </p>
        </div>
        <footer>
          

          <button
            className="btn btn-secondary"
            data-dialog-close="demo-dialog"
            type="button"
          >
            Cancel
          </button>
          

          <button
            className="btn btn-danger"
            type="button"
          >
            Delete
          </button>
          

        </footer>
      </section>
    </div>
    

  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Dialog HowItWorks renders 1`] = `
<Fragment>
  <h2
    id="how-it-works"
  >
    How it works
  </h2>
  <p>
    A dialog remains hidden until it is triggered. Once it is open it overlays the website with a dark background and presents some options, usually confirming or cancelling an action. If you use a dialog it is assumed that you want to "force" the user to make a choice. Therefore the only way to close a dialog is to choose one of the presented actions or, if given, press the 
    <i
      className="material-icons d-inline-flex"
    >
      close
    </i>
     to close the dialog.
  </p>
  <p>
    Dialog boxes consist of a header, a body and a footer. In these you can put whatever you want, but some basic functionality should be included in all dialogs. The header should contain a short descriptive text for what the dialog-box does. You can include an icon with class 
    <Property
      value=".dialog-close"
    />
     if you wish to give your user a way to close the dialog box without choosing an option. The dialog body should contain the main content, usually just text confirming an action. In the footer you will find buttons to confirm or cancel the requested action.
  </p>
  <Alert
    type="warning"
  >
    <h4>
      Initializing dialog buttons
    </h4>
    To make a button open the dialog box you need to include 
    <Attribute
      data={true}
      name="dialog-open"
      value="{#your-dialog-id}"
    />
     as an attribute to that button.
  </Alert>
</Fragment>
`;

exports[`Documentation: Dialog UsageWithJavascript renders 1`] = `
<Fragment>
  <h2
    id="usage-with-javascript"
  >
    Usage With Javascript
  </h2>
  <p>
    With the Dialog object you are given access to 
    <Property
      value="init()"
    />
    , 
    <Property
      value="open()"
    />
     and  
    <Property
      value="close()"
    />
    .
  </p>
  <InitDocumentation
    componentName="Dialog"
  />
  <h3>
    Toggling dialogs programmatically
  </h3>
  <p>
    We only support having one dialog open at-a-time, so 
    <Property
      value="dialog.open()"
    />
     and 
    <Property
      value="dialog.close()"
    />
     only accept single IDs as arguments.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="javascript"
  >
    px.dialog.open("demo-dialog");

    px.dialog.close("demo-dialog");

  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Dialog renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Dialog is meant to be used to show a snippet of text like a confirmation box before you delete something. Use 
    <Link
      to="/docs/components/sheet"
    >
      sheet
    </Link>
     if you wish to receive input from the user or display a large chunk of information.
  </p>
  <HowItWorks />
  <Example />
  <UsageWithJavascript />
</DocContainer>
`;
